enabler/test/de/schildbach/pte/live/VrsProviderLiveTest.java

```diff
@@ -0,0 +1,442 @@
+	}
+
+	@Test
+	public void testTripWithSurchargeInfo() throws Exception {
+		final QueryTripsResult result = queryTrips(new Location(LocationType.STATION, "687"), null, new Location(LocationType.STATION, "892"), new Date(), true, Product.ALL, WalkSpeed.NORMAL, Accessibility.NEUTRAL);
+		print(result);
+		assertEquals(QueryTripsResult.Status.OK, result.status);
+		assertTrue(result.trips.size() > 0);
+	}
+
+	@Test
+	public void manyRandomTrips() throws Exception {
+		Random rand = new Random(new Date().getTime());
+		int LAT_FROM = 50500000;
+		int LAT_TO = 51600000;
+		int LON_FROM = 6200000;
+		int LON_TO = 7600000;
+		int errors = 0;
+		long startTime = System.currentTimeMillis();
+		for (int i = 0; i < 3; i++) {
+			try {
+				int fromLat = LAT_FROM + rand.nextInt(LAT_TO - LAT_FROM);
+				int fromLon = LON_FROM + rand.nextInt(LON_TO - LON_FROM);
+				int toLat = LAT_FROM + rand.nextInt(LAT_TO - LAT_FROM);
+				int toLon = LON_FROM + rand.nextInt(LON_TO - LON_FROM);
+				final QueryTripsResult result = queryTrips(new Location(LocationType.ANY, fromLat, fromLon), null,
+						new Location(LocationType.ANY, toLat, toLon), new Date(), true, Product.ALL, WalkSpeed.NORMAL,
+						Accessibility.NEUTRAL);
+				System.out.println("# " + (i+1));
+				if (result.status.equals(QueryTripsResult.Status.OK)) {
+					print(result);
+				} else {
+					System.out.println("Status is " + result.status);
+					errors++;
+				}
+			} catch (SocketTimeoutException ex) {
+				System.out.println("SocketTimeoutException: " + ex);
+				errors++;
+			}
+		}
+		long stopTime = System.currentTimeMillis();
+		long elapsedTime = stopTime - startTime;
+		System.out.println("Elapsed: " + (elapsedTime / 1000) + " seconds");
+		System.out.println("Errors: " + errors);
+	}
+
+	private void crawlStationsAndLines(int LAT_FROM, int LAT_TO, int LON_FROM, int LON_TO) throws Exception {
+		Set<Location> stations = new TreeSet<Location>();
+		Random rand = new Random(new Date().getTime());
+		for (int i = 0; i < 5; i++) {
+			int lat = LAT_FROM + rand.nextInt(LAT_TO - LAT_FROM);
+			int lon = LON_FROM + rand.nextInt(LON_TO - LON_FROM);
+			System.out.println(i + " " + lat + " " + lon);
+			NearbyLocationsResult result = queryNearbyLocations(EnumSet.of(LocationType.STATION), new Location(LocationType.STATION, lat, lon), 0, 1);
+			stations.addAll(result.locations);
+		}
+		Set<Line> lines = new TreeSet<Line>();
+		for (Location station : stations) {
+			QueryDeparturesResult qdr = queryDepartures(station.id, false);
+			if (qdr.status == QueryDeparturesResult.Status.OK) {
+				for (StationDepartures sds : qdr.stationDepartures) {
+					for (LineDestination ld : sds.lines) {
+						lines.add(ld.line);
+					}
+				}
+			}
+		}
+
+		for (Location station : stations) {
+			System.out.println(station.toString());
+		}
+		for (Line line : lines) {
+			System.out.println(line.toString());
+		}
+	}
+
+	@Test
+	public void crawlStationsAndLinesNRW() throws Exception {
```

Comment:
Actually I'm not sure if we should add code to crawl. I'd like to avoid this code to be executed by (potentially many) CI servers, effectively DDoSing VRS server. At the very least, I think these tests should be set to @Ignore.